pvlib/irradiance.py

```diff
@@ -1147,17 +1147,17 @@ def perez(surface_tilt, surface_azimuth, dhi, dni, dni_extra,
     F2c = np.vstack((F2c, nans))
 
     F1 = (F1c[ebin, 0] + F1c[ebin, 1] * delta + F1c[ebin, 2] * z)
-    F1 = np.maximum(F1, 0)
+    F1 = np.where(np.isnan(F1), np.nan, np.maximum(F1, 0))
 
     F2 = (F2c[ebin, 0] + F2c[ebin, 1] * delta + F2c[ebin, 2] * z)
-    F2 = np.maximum(F2, 0)
+    F2 = np.where(np.isnan(F2), np.nan, np.maximum(F2, 0))
```

Comment:
if this polite handling is going to become a pattern, it might be worth putting it into `tools` as a dedicated set of functions like

```python
def polite_handling_of_nan(x, f):
    """return nan where x is nan otherwise return f(x)"""
    return np.where(np.isnan(x), np.nan, f(x))
```

Also isn't another option to turn off NumPy warnings? If that's an alternate, I'm curious if there is any penalty for calling `np.where()` all the time, instead of just ignoring the warnings.

Reply:
Also isn't another option to turn off NumPy warnings? If that's an alternate, I'm curious if there is any penalty for calling `np.where()` all the time, instead of just ignoring the warnings.